From 017147cfb6f8fb5c4f561e5090e6455fde2bc1af Mon Sep 17 00:00:00 2001
From: Tanmay Jagdale <tanmay@marvell.com>
Date: Fri, 6 Aug 2021 15:38:58 +0530
Subject: [PATCH 1897/1921] coresight: tmc: secure-etr: Fix incorrect offset

Fix the incorrect offset value passed during SMC call
to copy buffer from secure world to Linux DMA buffer.

Previously passed offset was incorrect when the buffer
was full and results in a copy operation beyond buffer
boundary which also causes an EL3 exception.

Fix this by passing 0x0 as offset since we always start
copying from the start of buffer and vary length field
(etr_buf->len) accordingly based on buffer full status.

Change-Id: I1866b558f0a3224b61d44961fd31bc6ea9e3e5af
Signed-off-by: Tanmay Jagdale <tanmay@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/58716
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Linu Cherian <lcherian@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/hwtracing/coresight/coresight-tmc-secure-etr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/hwtracing/coresight/coresight-tmc-secure-etr.c b/drivers/hwtracing/coresight/coresight-tmc-secure-etr.c
index b5141fc8bb2f..60849e91c546 100644
--- a/drivers/hwtracing/coresight/coresight-tmc-secure-etr.c
+++ b/drivers/hwtracing/coresight/coresight-tmc-secure-etr.c
@@ -366,7 +366,7 @@ static void tmc_etr_sync_secure_buf(struct etr_buf *etr_buf, u64 rrp, u64 rwp)
 	 * to copy the data to sysfs or perf buffer, we do not
 	 * generate SMC calls at different offsets everytime.
 	 */
-	tmc_copy_secure_buffer(secure_buf, etr_buf->offset, etr_buf->len);
+	tmc_copy_secure_buffer(secure_buf, 0x0, etr_buf->len);
 }
 
 static ssize_t tmc_etr_get_data_secure_buf(struct etr_buf *etr_buf,
-- 
2.31.1

